extension/chrome/settings/modules/contacts.ts

```diff
@@ -23,6 +33,48 @@ Catch.try(async () => {
 
   BrowserMsg.listen(tabId); // set_css
 
+  const attUI = new AttUI(() => Promise.resolve({ size_mb: 5, size: 5 * 1024 * 1024, count: 1 }));
+  attUI.initAttDialog('fineuploader', 'fineuploader_button');
+  attUI.setAttAddedCb(async (file: Att) => {
+    (file !== null && typeof file !== 'undefined' && typeof file.id !== 'undefined' && file.id !== '') ? attUI.clearAtt(file.id) : attUI.clearAtts();
+    await handleKeyfileSelected(file);
+  });
+
+  const handleKeyfileSelected = async (file: Att) => {
+    try {
+      const rawFile = file.getData();
+      const stringFile = new TextDecoder("utf-8").decode(rawFile);
+      const { blocks } = await Pgp.armor.detectBlocks(stringFile);
+
+      const keys: OpenPGP.key.Key[] = [];
+      if (blocks !== null && blocks.length > 0 && blocks.some(block => block.type === "publicKey")) {
```

Comment:
Check the return type, `blocks` will never be null - no need to check for it.

Reply:
Ok. Its a habit, but if you would prefer I ignore null checks when the typings don't provide a null I will ignore them. It will make my code more concise for sure.

Reply:
TypeScript is quite good with null and undefined types. If the type is wrong, I would rather that it throws an exception, so that we know to update the type.